pysatNASA/instruments/methods/cdaweb.py

```diff
@@ -568,6 +577,52 @@ def download(date_array, data_path, tag='', inst_id='', supported_tags=None,
                                   date.strftime('%d %B %Y'))))
         # Pause to avoid excessive pings to server
         sleep(0.2)
+
+    if zip_method:
+        # Cleanup temporary directory
+        temp_dir.cleanup()
+
+    return
+
+
+def get_file(remote_file, data_path, fname, temp_path=None, zip_method=None):
+    """Retrieve a file, unzipping if necessary.
+
+    Parameters
+    ----------
+    remote_file : file content
+        File content retireved via requests.
+    data_path : str
+        Path to pysat archival directory.
+    fname : str
+        Name of file on the remote server.
+    temp_path : str
+        Path to temporary directory. (Default=None)
+    zip_method : str
+        The method used to zip the file. Supports 'zip' and None.
+        If None, downloads files directly. (default=None)
+
+    """
+
+    if zip_method:
+        # Use a temporary location.
+        dl_fname = os.path.join(temp_path, fname)
+    else:
+        # Use the pysat data directory.
+        dl_fname = os.path.join(data_path, fname)
+
+    # Download the file to desired destination.
+    with open(dl_fname, 'wb') as open_f:
+        open_f.write(remote_file)
+
+    # Unzip and move the files from the temporary directory.
+    if zip_method == 'zip':
+        with zipfile.ZipFile(dl_fname, 'r') as open_zip:
+            open_zip.extractall(data_path)
+
+    elif zip_method is not None:
+        logger.warning('{:} is not a recognized zip method'.format(zip_method))
```

Comment:
Recommend adding a test for this warning.  Also creating an issue for the future tests, if it hasn't been done already.

Reply:
documented as #213